src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java

```diff
@@ -334,7 +335,7 @@ public PrivilegeSummary getLatestBookingIEPSummaryForOffender(@NotNull @PathVari
     @ApiOperation("Return a list of damage obligations")
     @GetMapping("/{offenderNo}/damage-obligations")
     public OffenderDamageObligationResponse getOffenderDamageObligations(@NotNull @PathVariable("offenderNo") @ApiParam(value = "offenderNo", required = true, example = "A1234AA") final String offenderNo, @RequestParam(value = "status", required = false, defaultValue = "") @ApiParam(value = "Filter by obligation status. Leave blank to return all", required = false, example = "ACTIVE", allowableValues = "INACT,PAID,ONH,ACTIVE,APPEAL") final String status) {
-        final var damageObligations = offenderDamageObligationService.getDamageObligations(offenderNo, status);
+        final var damageObligations = offenderDamageObligationService.getDamageObligations(offenderNo, Status.forCode(status));
```

Comment:
This could be replace with Status.valueOf has the Enum type has this functionality built in by default. 

Also, one for another PR but you could add a Converter so that spring takes care for deserializing for you.

Reply:
the forCode has custom logic which valueOf doesn't offer
   ```
      public static Status forCode(String code) {
            return Arrays.stream(Status.values())
                .filter(c -> c.equals(code.toUpperCase()))
                .findFirst()
                .orElse(ALL);
        }
```
how would valueOf default to ALL for an empty string?

Reply:
done

Reply:
One way would to set a default value for the controller status parameter. Then you would just use Status.valueOf(status).

Reply:
For example just set the default to 'ALL'

Reply:
done